### ISSUE TITLE
redundant_closure false positive

### ISSUE THREAD

**Issue by frankmcsherry** [rust-lang-nursery/rust-clippy#2024]
This didn't seem to be a repeat, but was a different flavor of false positive. In this case, I'm using a closure to wrap an element in an enum variant, `Local`. Clippy seems to interpret `|x| Local(x)` as "calling the function `Local`".

```
warning: redundant closure found
  --> src/progress/nested/summary.rs:84:104
   |
84 |             (&Local(ref inner1), &Local(ref inner2))                 => inner1.followed_by(inner2).map(|x| Local(x)),
   |                                                                                                        ^^^^^^^^^^^^ help: remove closure as shown: `Local`
   |
   = note: #[warn(redundant_closure)] on by default
   = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/0.0.157/index.html#redundant_closure
```

**Comment by oli-obk** [comment]
That's not a false positive. `map(Local)` will work
